# Please do not modify this file, this configuration is the default configuration
# if you need to modify, please modify xarm_user_params.yaml
ufactory_driver:
  ros__parameters:
    joint_names: ['joint1', 'joint2', 'joint3', 'joint4', 'joint5', 'joint6', 'joint7']
    xarm_gripper:
      joint_names: ['drive_joint', 'left_finger_joint', 'left_inner_knuckle_joint', 'right_outer_knuckle_joint', 'right_finger_joint', 'right_inner_knuckle_joint']
      speed: 2000
      max_pos: 850
      frequency: 10
      threshold: 3
      threshold_times: 10
    services:
      debug: false  # When debug is true, all services will be turned on
      clean_error: true
      clean_warn: true
      clean_conf: false
      save_conf: false
      reload_dynamics: false
      set_counter_reset: false
      set_counter_increase: false
      clean_gripper_error: false
      clean_bio_gripper_error: false
      start_record_trajectory: false
      stop_record_trajectory: false
      get_state: false
      get_cmdnum: false
      get_vacuum_gripper: false
      get_gripper_err_code: false
      get_bio_gripper_status: false
      get_bio_gripper_error: false
      get_err_warn_code: false
      set_mode: true
      set_state: true
      set_collision_sensitivity: false
      set_teach_sensitivity: false
      set_gripper_mode: false
      set_gripper_enable: false
      set_tgpio_modbus_timeout: false
      set_bio_gripper_speed: false
      set_collision_rebound: false
      set_fence_mode: false
      set_reduced_mode: false
      set_self_collision_detection: false
      set_simulation_robot: false
      set_baud_checkset_enable: false
      motion_enable: true
      set_servo_attach: false
      set_servo_detach: false
      set_reduced_tcp_boundary: false
      get_tgpio_modbus_baudrate: false
      set_tgpio_modbus_baudrate: false
      get_checkset_default_baud: false
      set_checkset_default_baud: false
      get_gripper_position: false
      get_position: false
      get_servo_angle: true
      get_position_aa: false
      set_pause_time: false
      set_tcp_jerk: false
      set_tcp_maxacc: false
      set_joint_jerk: false
      set_joint_maxacc: false
      set_gripper_speed: false
      set_reduced_max_tcp_speed: false
      set_reduced_max_joint_speed: false
      set_gravity_direction: false
      set_tcp_offset: false
      set_world_offset: false
      set_reduced_joint_range: false
      set_tcp_load: false
      set_position: true
      set_tool_position: false
      set_position_aa: false
      set_servo_cartesian: false
      set_servo_cartesian_aa: false
      set_servo_angle: true
      set_servo_angle_j: true
      move_circle: false
      move_gohome: true
      vc_set_joint_velocity: true
      vc_set_cartesian_velocity: true
      get_tgpio_digital: false
      get_cgpio_digital: false
      get_tgpio_analog: false
      get_cgpio_analog: false
      set_tgpio_digital: false
      set_cgpio_digital: false
      set_tgpio_digital_with_xyz: false
      set_cgpio_digital_with_xyz: false
      set_cgpio_analog: false
      set_cgpio_analog_with_xyz: false
      set_vacuum_gripper: false
      set_gripper_position: false
      set_bio_gripper_enable: false
      open_bio_gripper: false
      close_bio_gripper: false
      robotiq_reset: false
      robotiq_set_activate: false
      robotiq_set_position: false
      robotiq_open: false
      robotiq_close: false
      robotiq_get_status: false
      getset_tgpio_modbus_data: false
      save_record_trajectory: false
      load_trajectory: false
      playback_trajectory: false